### DISCUSSION TITLE
want to know if child component of client component , is convert automatically into client if parent is client ?

### DISCUSSION THREAD

**Discussion by jaimeenkoffeekodes** [vercel/next.js#61842] [Help] [answered]
### Summary

my root template component is client component, so my question is that , children of this will be considered client or server ? 
for reference you can take this code (template.tsx): 

### Additional information

"use client";
import { usePathname } from "next/navigation";
import BreadCrumbs from "./_components/BreadCrumbs";
import HeaderForDashboard from "./_components/server_components/HeaderForDashboard";
import Sidebar from "./_components/server_components/Sidebar";
import Header from "./_components/server_components/Header";
import { useContext } from "react";
import { DesignContext } from "./_context/DesignContext";

export default function Template({ children }: { children: React.ReactNode }) {
  const pathname = usePathname();
  const { sidebarToggle } = useContext(DesignContext);

  const PathWithoutLayout = ["/login", "/loader", "/loader2"];
  if (PathWithoutLayout.includes(pathname)) {
    return <div>{children} </div>;
  }
  const companyLayout = [
    "/dashboard",
    "/master",
    "/gst",
    "/account_book",
    "/other_book",
    "/reporting",
    "/stock",
    "/transactions",
  ];
  if (companyLayout.includes(pathname)) {
    return (
      <div className="flex h-auto scrollbar-remove" id="dashboard">
        <div className="fixed top-0 left-0 bottom-0 bg-white">
          <Sidebar />
        </div>
        <div
          className={`fixed top-0 right-0 left-0 z-10 bg-[--background] ${
            !sidebarToggle ? "md:ml-[66px]" : "md:ml-[232px]"
          }`}
        >
          <HeaderForDashboard />
        </div>

        <div className="h-auto bg-[--background] flex-grow">
          <div
            className={`${
              !sidebarToggle ? "md:ml-[80px]" : "md:ml-[244px]"
            } mt-[80px] scrollbar-remove min-h-screen overflow-y-scroll bg-[--background]`}
          >
            <BreadCrumbs subMenu="dashboard" />
            {children}
          </div>
        </div>
      </div>
    );
  }
  return (
    <div className="min-h-screen bg-[--background]">
      <Header />
      <div className="mx-5 scrollbar-remove overflow-y-scroll">
        {/* <div className="h-[calc(100%-170px)] scrollbar-remove overflow-y-scroll mx-5 border bg-white shadow-md"> */}
        {children}
      </div>
    </div>
  );
}

**Comment by icyJoseph** [comment, marked as answer]
Hi,

To be more precise, imports, both direct and indirect are opted into client modules (sent to the browser). You can think in terms of ownership. 

`BreadCrumbs`, `HeaderForDashboard`, `Sidebar`, etc... are all owned by this `template.tsx` file, and since `template.tsx` has the `use client` label, `BreadCrumbs`, `HeaderForDashboard`, `Sidebar`, etc... will be client modules...

However! The `children` prop, is not owned by the Template component, even though, in the hierarchy, Template is the parent! That means that when Template is used, it can have Server Component children!

```tsx
<Template>
   <TotallyAServerComponent />
</Template>
```

The above is a valid pattern!

**Reply by jaimeenkoffeekodes** [reply]
Thank you so much for clarifying that! Your explanation was incredibly helpful.
